maths/juggler_sequence.ts

```diff
@@ -0,0 +1,27 @@
+/**
+ * The juggler sequence is a integer sequence that starts with an positive integer a and the subsequent terms are
+ * described as following:
+ * if a_k is even:
+ *   a_k+1 = floor(sqrt(a_k))
+ * else:
+ *   a_k+1 = floor(sqrt(a_k^3))
+ *
+ * Time Complexity: linear (O(n))
+ *
+ * @param a The number to start with
+ * @param n The index of the searched number in the sequence.
+ * @returns The number at index n in the sequence.
+ * @see https://en.wikipedia.org/wiki/Juggler_sequence
+ */
+export const jugglerSequence = (a: number, n: number) => {
+  let k: number = a;
+  for (let i: number = 0; i < n; i++) {
+    if (k % 2 === 0) {
```

Comment:
Could even shorten this to `k = Math.floor(Math.pow(k, (1 + 2*(k % 2)) / 2))` (granted, this is perhaps slightly unreadable) or, if you prefer a ternary, `k = Math.floor(Math.pow(k, (k % 2 === 0 ? 1 : 3) / 2))`.

Reply:
I agree, but I would prefer to leave it like it is for the sake of readability and to make it easier to understand for people who read this code.

Reply:
I think the ternary version is readable. But it's fine as-is too.

Reply:
Oh yes, I don't know how, but I haven't noticed the ternary version in your comment. Yeah, the ternary version looks applicable and as easy to understand as the current version. I will add this change real quick.